### ISSUE TITLE
Inquiry Regarding dataset_name + "_filenames.txt" Files in Your Code

### ISSUE THREAD

**Issue by BlouseDong** [akhilpm/DroneDetectron2#41]
I am currently in the process of reproducing your code. During this process, I encountered a question regarding the dataseed folder in your codebase, which contains files named dataset_name + "_filenames.txt".
I am unsure about the purpose and generation method of these files and would greatly appreciate your guidance on the matter.

1.Are these dataset_name + "_filenames.txt" files meant to be manually created by me, or does your code generate them automatically?
2.If I am to create them manually, what content should they contain? For instance, should they include image file names or file paths? What is the expected format of the content?
3.If they are generated by the code, could you please direct me to the script responsible for their generation? Additionally, how should I go about running this script?

Thank you very much for your patience and assistance. I am looking forward to your response.
Best regards,

**Comment by akhilpm** [comment]
It is a utility function I used to split the training set into two for running experiments quickly. Ideally, you don't need to do this splitting if you are running your training on the full set of images. So, you can comment the following line

[https://github.com/akhilpm/DroneDetectron2/blob/main/croptrain/data/build.py#L74](https://github.com/akhilpm/DroneDetectron2/blob/main/croptrain/data/build.py#L74)

If you want to create that file, just get the filename of each image from the annotations_dataset.json.

**Comment by BlouseDong** [comment]
Thank you for your response. I divided label_dicts and unlabel_dicts into equal halves based on the train.json. However, when running into dataset_dict['full_image'] in detection_utils.py, it raised a keyword error. I'm not quite sure where you added these keywords that were originally not in the dataset annotation files. Also, the keyword "two_stage_crop" in dataset_dict["two_stage_crop"] is unfamiliar to me.

Below is an example of one of your dataset_dict entries:
{
    'file_name': 'D:\\dataset\\mycoco/train\\9999953_00000_d_0000217.jpg', 
    'height': 788, 
    'width': 1400, 
    'image_id': 2884, 
    'annotations': [
        {'iscrowd': 0, 'bbox': [246, 565, 50, 29], 'category_id': 8, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [263, 513, 54, 27], 'category_id': 3, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [627, 450, 31, 55], 'category_id': 3, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [709, 422, 21, 38], 'category_id': 3, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [658, 309, 20, 35], 'category_id': 3, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [680, 177, 21, 41], 'category_id': 3, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [700, 60, 26, 41], 'category_id': 3, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [498, 204, 7, 14], 'category_id': 0, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [521, 207, 7, 21], 'category_id': 0, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [520, 201, 8, 20], 'category_id': 0, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}, 
        {'iscrowd': 0, 'bbox': [472, 183, 14, 21], 'category_id': 0, 'bbox_mode': <BoxMode.XYWH_ABS: 1>}
    ]
}

**Comment by BlouseDong** [comment]
i used RCNN-FPN-CROP.yaml

**Comment by akhilpm** [comment]
I added this in the dataset dict creation stage. It will be set automatically when you run my script, I don't know what modifications you did which resulted in the absence of them. 

"two_stage_crop" flag has impact only for DOTA dataset. Because DOTA images are 4k+ in resolution, I did the training on patches from the original image. "two_stage_crop" is used to identify density crops obtained from these patches (two inner levels).
